actors/market/tests/on_miner_sectors_terminate.rs

```diff
@@ -166,57 +227,35 @@ fn terminate_valid_deals_along_with_expired_and_cleaned_up_deal() {
     let deal_ids = publish_deals(
         &rt,
         &MinerAddresses::default(),
-        &[deal1, deal2.clone()],
+        &[deal1.clone(), deal2.clone()],
         TokenAmount::zero(),
         1,
     );
     assert_eq!(2, deal_ids.len());
-    activate_deals(&rt, sector_expiry, PROVIDER_ADDR, current_epoch, &deal_ids);
+    activate_deals_legacy(&rt, sector_expiry, PROVIDER_ADDR, current_epoch, &deal_ids);
 
     let new_epoch = end_epoch - 1;
     rt.set_epoch(new_epoch);
     cron_tick(&rt);
-
-    terminate_deals(&rt, PROVIDER_ADDR, &deal_ids);
-    assert_deals_terminated(&rt, new_epoch, &deal_ids[0..0]);
+    // expired deal deleted normally
     assert_deal_deleted(&rt, deal_ids[1], deal2);
-    check_state(&rt);
-}
+    assert_deals_not_terminated(&rt, &deal_ids[0..0]);
 
-// Converted from: https://github.com/filecoin-project/specs-actors/blob/d56b240af24517443ce1f8abfbdab7cb22d331f1/actors/builtin/market/market_test.go#L1369
-#[test]
-fn terminating_a_deal_the_second_time_does_not_change_its_slash_epoch() {
-    let start_epoch = 10;
-    let end_epoch = start_epoch + 200 * EPOCHS_IN_DAY;
-    let sector_expiry = end_epoch + 100;
-    let current_epoch = 5;
+    terminate_deals_and_assert_balances(&rt, CLIENT_ADDR, PROVIDER_ADDR, &deal_ids);
+    // terminated deal deleted
+    assert_deal_deleted(&rt, deal_ids[0], deal1);
 
-    let rt = setup();
-    rt.set_epoch(current_epoch);
-
-    let deal1 = generate_and_publish_deal(
+    // terminated deal has a dangling deal op, normally expired deal doesn't
+    check_state_with_expected(
         &rt,
-        CLIENT_ADDR,
-        &MinerAddresses::default(),
-        start_epoch,
-        end_epoch,
+        &[Regex::new("^deal op found for deal id \\d+ with missing proposal at epoch \\d+$")
+            .unwrap()],
     );
-    activate_deals(&rt, sector_expiry, PROVIDER_ADDR, current_epoch, &[deal1]);
-
-    // terminating the deal so slash epoch is the current epoch
-    terminate_deals(&rt, PROVIDER_ADDR, &[deal1]);
-
-    // set a new epoch and terminate again -> however slash epoch will still be the old epoch.
-    rt.set_epoch(current_epoch + 1);
-    terminate_deals(&rt, PROVIDER_ADDR, &[deal1]);
-    let s = get_deal_state(&rt, deal1);
-    assert_eq!(s.slash_epoch, current_epoch);
-    check_state(&rt);
 }
 
-// Converted from: https://github.com/filecoin-project/specs-actors/blob/d56b240af24517443ce1f8abfbdab7cb22d331f1/actors/builtin/market/market_test.go#L1387
+// Converted from: https://github.com/filecoin-project/specs-actors/blob/d56b240af24517443ce1f8abfbdab7cb22d331f1/actors/builtin/market/market_test.go#L1369
```

Comment:
Delete these "converted from" comments as you encounter them. We don't need them any more.